src/creature.cpp

```diff
@@ -211,11 +211,11 @@ void Creature::onWalk()
 				forceUpdateFollowPath = true;
 			}
 		} else {
+			stopEventWalk();
+
 			if (listWalkDir.empty()) {
 				onWalkComplete();
 			}
-
-			stopEventWalk();
```

Comment:
Walk should actually be completed (`onWalkCompleted`) after the event is stopped. 🙄
This also makes it possible for monster to walk by smaller paths.